packages/expo-dev-launcher/src/DevLauncher.ts

```diff
@@ -0,0 +1,48 @@
+import { ErrorUtils } from 'react-native';
+
+// Similar interface to the one used in expo modules.
+type CodeError = Error & { code?: string };
+
+let errorHandlerWasRegistered = false;
+
+function customizeModuleIsMissingMessage(error: Error) {
+  error.message += `
```

Comment:
Could we completely overwrite the error message instead of leaving in the vague "Native module cannot be null"? I would lean towards something like this:

```
Your JavaScript code tried to access a native module that doesn't exist in this development client. Some possible solutions:
- Make sure you're using the newest available version of this development client.
- Make sure you're running the latest published update of your JavaScript code.
- If you've installed a new library recently, you may need to make a new development client build.
```

also cc @cruzach in case you have an opinion about what would be most helpful to say here.

Reply:
+1 to not showing the vague and unhelpful "Native module cannot be null"

one small change i would suggest to the error message you shared:

```
Your JavaScript code tried to access a native module (${NATIVE-MODULE-NAME}) that doesn't exist in this development client. Some possible solutions:
```

For:

```
Make sure you're running the latest published update of your JavaScript code.
```

is "latest" the term we want to use? With multiple people working on an app it's possible that the most recently published bundle is not the one they'd want to use, right? Maybe "Make sure you're running a compatible version of your JavaScript code." or similar

Reply:
We can't get the module name from the exception. 

Here you can see an exception. From that message and stack trace, you won't guess it's caused by the missing Reanimated 🤷
![image](https://user-images.githubusercontent.com/9578601/120987740-2d9cf580-c77e-11eb-859f-97f399a769de.png)

So I've replaced the `Native module cannot be null` error message with your proposition, but I didn't replace the message in the `CodedError` case.